Result/Result.swift

```diff
@@ -114,18 +114,30 @@ public enum Result<Value, Error: Swift.Error>: ResultProtocol, CustomStringConve
 	}
 }
 
+extension Result where Error == AnyError {
+	/// Constructs a result from an expression that uses `throw`, failing with `AnyError` if throws.
+	public init(_ f: @autoclosure () throws -> Value) {
+		self.init(attempt: f)
+	}
+
+	/// Constructs a result from a closure that uses `throw`, failing with `AnyError` if throws.
+	public init(attempt f: () throws -> Value) {
+		do {
+			self = .success(try f())
+		} catch {
+			self = .failure(AnyError(error))
+		}
+	}
+}
```

Comment:
I've just added https://github.com/antitypical/Result/pull/236/files#diff-edf3d2a3570d3672e50cbe6ca3ede069R140 and without these overloads that does not compile.

The test case / result are bit different from #233.